toolchain/check/import_ref.cpp

```diff
@@ -1238,8 +1239,11 @@ template <typename InstT>
 static auto ResolveAsUnique(ImportRefResolver& resolver,
                             SemIR::InstId import_inst_id, InstT inst)
     -> ResolveResult {
-  static_assert(InstT::Kind.constant_kind() == SemIR::InstConstantKind::Unique,
-                "Use ResolveAsDeduplicated");
+  static_assert(
+      InstT::Kind.constant_kind() == SemIR::InstConstantKind::Unique ||
+          InstT::Kind.constant_kind() ==
+              SemIR::InstConstantKind::ConditionalUnique,
+      "Use ResolveAsDeduplicated");
```

Comment:
Could this be a `requires` instead?

Reply:
Yes, it could be, but we can't provide a custom error message so easily that way. Are you thinking that we give the two `ResolveAs...` functions the same name again? If we keep them separate, a diagnostic that mentions the other function name seems at least a little valuable.

Reply:
Maybe I'm just ambivalent about the `ResolveAsDeduplicated`, on two ends:

- I think people will figure it out from a simple compile error, a comment seems equivalent (this isn't a high-use API).
- I don't see why the name changed, when `requires` could have achieved equivalent enforcement. But, I didn't suggest/review it so I don't want to push too much.

I mainly noted it, though, since I've recently bumped into issues with static_assert and having to migrate things to requires. :)

Reply:
But note, I also am hesitant to just churn code when I don't see a benefit. It's not like I think overloading would really fix an issue. I just am losing my love for static_assert when requires would suffice. :)

Reply:
Switched to a concept.